### ISSUE TITLE
Unable to update from 2.19.2 to 2.21.0

### ISSUE THREAD

**Issue by DevN1mr0d** [fetlife/android#517]
Old story, unable to update from app, never downloads the new version, keeps downloading endlessly. I know, out of scope bug, solved by downloading directly from Git.
However my version was 2.19.2 but there were several stable versions since then. The app has never notified me on updated versions, only by now regarding 2.21.0

**Comment by FernandoMiguel** [comment]
I was on the previous release, got the notification, it downloaded (could see the apk on the download folder), but it never install when pressing the notification ready to install.

Downloading from github installed fine

**Comment by calvinwiii** [comment]
Install totally fails if app is in secure folder. Install gives notification and infinite download message if in normal folder, but won't install. Can the be manually installed from the download folder, so the download did happen. After update, fetlife still indicates update available and retries

**Comment by DevN1mr0d** [comment]
Downloading package from Git works perfectly, but issue #516 also still exists.

**Comment by kistitan** [comment]
Hi guys,

We are unfortunately aware of the several installation issues, however after several reviews, our code looks fine, as we transfer the download and installation process to the built in download manager of the phone. 
From that point all the download and installation is not in our hand anymore.

We are thinking of other options, and keep investigating what can we do...

**Comment by kdom13** [comment]
I've been having this issue as well.  I'm currently on 2.19.2, and have gotten update notifications at least twice (maybe three times), but it won't actually update.  I haven't had a chance to do it manually yet, I'll update now.

**Comment by FernandoMiguel** [comment]
The lasted also downloaded fine but failed to install
Manual install worked

**Comment by kistitan** [comment]
Hey guys,

Having a small update about the recent notification issues (still some others that we do not see reasons for).
Recently we realized that if there is no stable and preview release available the same time, older versions of the App misses the notifications, however in case having both of them, newer versions can get confused.
So we released the fix for that, but had to make a choice between two not very nice options, but as some of the users with old versions did not get notified about updates for a while we chose to help them out a bit, so we released the same version as stable and as preview version.
This could cause some issues in notifications, and that's why it might have seemed the App tries to download the same version, however it was a different file.

We apologize for the confusion, hopefully with the latest update at least this issue won't cause more problems.
We still consider to try find a better update process, as all the external apps we rely on like now (default download manager, guided installation process, download apk) seems to be fragile on some devices. These are unfortunately out of our hands, but probably there are still some things we can improve in the future, like retry mechanism, clear instructions, etc.

Thanks for your help and patience
